lib/superintendent/request/params_case_translator.rb

```diff
@@ -27,10 +27,31 @@ def underscored_keys(value)
       when Array
         value.map { |v| underscored_keys(v) }
       when Hash
+        if value['data'].present?
+          transform_type_to_snake_case(value['data'])
+        end
```

Comment:
That works and I like it. But it will change any `type` property it finds. I'm fine with that, but I was under the impression that we need to touch only the type properties that have a `data` property as parent. If this is not the case, I can gladly change the PR to contain the code proposed by @spencerroan.